src/types/resource/stream.rs

```diff
@@ -149,9 +180,9 @@ impl Stream {
                 match url.path_segments_mut() {
                     Ok(mut path) => {
                         path.push(&hex::encode(info_hash));
-                        if let Some(file_idx) = file_idx {
-                            path.push(&file_idx.to_string());
-                        }
+                        path.push(
+                            &file_idx.map_or_else(|| "-1".to_string(), |idx| idx.to_string()),
+                        );
```

Comment:
Could you once more document when we use `-1` as value. As someone that's still relatively new to the project, I'm not aware of all these options and would be nice for future contributors to also have this knowledge in the source code